### PR TITLE
update cargo to 0.11, closes #171

### LINKED ISSUE #171: Update Hyper
and go full async

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -54,12 +54,11 @@ pub fn run_server(build_args: Option<BuildArgs>) {
     let ip = config.ip.clone();
     let port = config.port;
 
-    let server = server::Instance::new(config, build_args);
-
     println!("server running on http://{}:{}", ip, port);
-    Server::http(&*format!("{}:{}", ip, port))
+    let addr = format!("{}:{}", ip, port).parse().unwrap();
+    Http::new().bind(&addr, move || Ok(server::Instance::new(config.clone(), build_args.clone())))
```
Comment (nrc) [inline src/lib.rs]: You shouldn't need to clone `config` and `build_args` - they aren't used later, so should be OK to move.

File: src/server.rs [hunk 2]
```diff
@@ -165,162 +165,141 @@ impl<'b> DiagnosticEventHandler<'b> {
         match parsed {
             errors::ParsedError::Diagnostic(d) => {
                 let text = serde_json::to_string(&d).unwrap();
-                self.res.write_all(format!("event: error\ndata: {}\n\n", text).as_bytes())
-                    .unwrap();
-                self.res.flush().unwrap();
+                self.res.set_body(format!("event: error\ndata: {}\n\n", text));
+                // self.res.set_body(format!("event: error\ndata: {}\n\n", text).as_bytes());
```
Comment (nrc) [inline src/server.rs]: comment should be removed

File: src/server.rs [hunk 3]
```diff
@@ -165,162 +165,141 @@ impl<'b> DiagnosticEventHandler<'b> {
         match parsed {
             errors::ParsedError::Diagnostic(d) => {
                 let text = serde_json::to_string(&d).unwrap();
-                self.res.write_all(format!("event: error\ndata: {}\n\n", text).as_bytes())
-                    .unwrap();
-                self.res.flush().unwrap();
+                self.res.set_body(format!("event: error\ndata: {}\n\n", text));
```
Comment (nrc) [inline src/server.rs]: I don't think it is correct to use `set_body` here - the old write and flush would append the message and send it to the client, whereas `set_body` will replace any old text and not send the data immediately (note that this is an event stream where we stream messages to the client).

File: src/lib.rs (hunk 1 above)
Comment (jxs) [reply src/lib.rs]: if I don't clone i get: ``` this closure implements `FnOnce`, not `Fn` ``` 
and if I don't move i get: 
 ``` closure may outlive the current function, but it borrows `config`, which is owned by the current function ```

**Comment by jxs** [comment]
updated! with the help from folks at #hyper i was able to update the solution, it was suggested to spawn a thread with the blocking computing and then `send` the output to the `Body`
might still need some review! 
cheers :)

**Comment by nrc** [comment]
Thanks for the update. It looks good to me. There are a couple of warnings from the compiler which look like they should be addressed though.

**Comment by jxs** [comment]
yeah, sorry forgot to remove the line that was triggering the `code unreachable` warning

**Comment by nrc** [comment]
Thanks!

**Pull request merged**